### ISSUE TITLE
Contributing guidelines

### ISSUE THREAD

**Issue by gbrivady** [iScsc/wiresmash#10]
Everything already included in the wiki [https://github.com/iScsc/wiresmash/wiki/Contributing](url)
- Conventions
- Instructions to add source file to a CMake target

TODO :
- [ ] specify workflow
- [ ] everything else I can't think of
- [ ] Link to the CONTRIBUTING.md

**Comment by gbrivady** [comment]
I have no clue what to add, so I ask for input from you : @amtoine @ctmbl @atxr

**Comment by ctmbl** [comment]
I added some changes and examples to the wiki: see [the diff](https://github.com/iScsc/wiresmash/wiki/Contributing/_compare/1150129f098ce26973f44047b4313821c1764c91...3016f42960b303f4c221127a143703055f78817d)

**Comment by gbrivady** [comment]
> Revert/update it without noticing me if you want 😉

Nope, good update 👌 

> struct are named `MyStruct_t` (the `_t` meaning type I think) in `C`, is it different in cpp?

Tbh I have no clue, but since enumerations, classes and structures are equivalent in Cpp (no need to use a typedef to remove the struct keyword for example), modulus some properties when it comes to access control (all struc members & attributes are public by default), I think it is for the best if we keep it simple.

> proper home page

I think I will work on this when the wiki is more "full" as for now I believe we will only add redudancy - so better keep a clean sheet;

> I can also think of adding the: fork, branch, commit, push, PR workflow (described as GitHub Flow in the website wiki) but maybe that will go in the CONTRIBUTING.md, other thing: when and how to add unit tests (but waybe wait for #6 )

Did not think about that! Will steal the workflow from iscsc to create the `CONTRIBUTING.md`.

> How about using a formater/linter?

Using one would be a good idea, will make another issue. Could also be useful to force formatting before every PR

I think I have the global view on where to "seed" information about contributing:

**Wiki**
- Conventions (naming, for code, file names... -> mostly already there)
- Detailled how-to for documenting
- Detailled how-to for code "ownership", ie the `@author` tag of doxygen
- Detailled workflow 
- Detailled information on CMake usage, doxygen usage, ... tools in general

And maybe a "why" on our decisions, so our thought process does not get lost over time;

**`CONTRIBUTING.md`** 
- TLDR of workflow
- link to the wiki for detailled information
- how to use the formatter - when we will have one available

**Comment by amtoine** [comment]
[Contributing](https://github.com/iScsc/wiresmash/wiki/Contributing) looks ok for now :thinking: 

next we should have a PR to add `CONTRIBUTING.md` :+1: 

i agree that #14 will enforce the style and some conventions, we should rely on automatic checks rather than just simple plain text guidelines for humans :+1:

**Comment by ctmbl** [comment]
@gbrivady I think the wiki page would greatly benefit from your wonderful UML diagram from #2 
Actually you could add nearly everything from this PR first message

And if my previous remarks has been addressed then let's go write a `CONTRIBUTING.md` to finally close this issue!
